libr/core/file.c

```diff
@@ -122,6 +122,10 @@ R_API int r_core_file_reopen(RCore *core, const char *args, int perm, int loadbi
 		// XXX - select the right backend
 		if (core->file && core->file->desc) {
 			newpid = core->file->desc->fd;
+#if __linux__
```

Comment:
this shouldnt be something specific for each backend... we may have the same problem on BSD ? (doesnt seems to fail on OSX)